tests/test_linux.py

```diff
@@ -274,3 +274,21 @@ def test_serialize_state_with_closed_files(self):
         fd = platform.sys_open(filename, os.O_RDONLY, 0o600)
         platform.sys_close(fd)
         pickle.dumps(platform)
+
+    def test_thumb_mode_entrypoint(self):
+        m = Manticore.linux(os.path.join(os.path.dirname(__file__), 'binaries', 'thumb_mode_entrypoint'))
+
+        @m.init
+        def init(state):
+            state.cpu.regfile.write('R0', 0)
+            state.cpu.regfile.write('R1', 0x1234)
+            state.cpu.regfile.write('R2', 0x5678)
+
+            @m.hook(None)
```

Comment:
Is this indentation here proper? Shouldn't it be one less level?

Reply:
also: is it good to assume the hook will fire? Maybe we should put a:

```python
hook_fired = False

@m.hook(None)
def success(state):
    # ...
    hook_fired = True

self.assertTrue(hook_fired)
```
?

Reply:
Yes, the indentation is as intended. But in retrospect I agree that it maybe isn't the easier to read. I will refactor the unit test a bit in my next commit.